libsolidity/interface/CompilerStack.cpp

```diff
@@ -1205,13 +1206,76 @@ Json CompilerStack::ethdebug(Contract const& _contract, bool _runtime) const
 	solAssert(m_stackState >= AnalysisSuccessful, "Analysis was not successful.");
 	solAssert(_contract.contract);
 	solUnimplementedAssert(!isExperimentalSolidity());
+	Json result = Json::object();
+	Json contract = Json::object();
+	contract["name"] = _contract.contract->name();
+	Json definition = Json::object();
+	definition["sources"]["id"] = static_cast<int>(sourceIndices()[_contract.contract->sourceUnitName()]);
+	contract["definition"] = definition;
+	result["contract"] = contract;
+	result["environment"] = _runtime ? "call" : "create";
+	result["instructions"] = ethdebugInstructions(_contract, _runtime);
+	return result;
+}
+
+Json CompilerStack::ethdebugInstructions(Contract const& _contract, bool _runtime) const
+{
+	evmasm::LinkerObject const* object;
+	evmasm::Assembly const* assembly;
 	if (_runtime)
 	{
-		Json result = Json::object();
-		return result;
+		object = &_contract.runtimeObject;
+		assembly = _contract.evmRuntimeAssembly.get();
 	}
-	Json result = Json::object();
-	return result;
+	else
+	{
+		object = &_contract.object;
+		assembly = _contract.evmAssembly.get();
+	}
+
+	Json instructions = Json::array();
+	for (size_t i = 0; i < object->offsets.size(); ++i)
+	{
+		size_t offset = object->offsets[i];
+		size_t nextOffset = offset;
+		if (i + 1 < object->offsets.size())
+			nextOffset = object->offsets[i + 1];
+		Json instruction = Json::object();
+		instruction["offset"] = offset;
+		instruction["mnemonic"] = instructionInfo(static_cast<evmasm::Instruction>(object->bytecode[offset]), m_evmVersion).name;
+		bytes argumentData;
+		for (size_t args = offset + 1; args < nextOffset; ++args)
+			argumentData.emplace_back(object->bytecode[args]);
+		if (!argumentData.empty())
+		{
+			Json arguments = Json::array();
+			arguments.emplace_back("0x" + util::toHex(argumentData));
+			instruction["arguments"] = arguments;
+		}
+
+		solAssert(assembly->codeSections().size() == 1);
```

Comment:
Why can there only be one?

Reply:
only eof can have more

Reply:
I just changed that to an `solUnimplementedAssert`. need to add a test.

Reply:
ok.. simple test added now